quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImpl.java

```
@@ -39,11 +40,16 @@ public Collection addCollection(String name, String description) {
      * 
      * 
      * @return Collection Instance that is found from the database
+     * @throws InvalidObjectIdException if collectionId couldn't be converted to ObjectId
      * 
      **/
     @Override
-    public Collection findCollection(String id) {
-        return collectionRepo.findById(new ObjectId(id)).orElse(null);
+    public Collection findCollection(String id) throws InvalidObjectIdException {
+        try {
+            return collectionRepo.findById(new ObjectId(id)).orElse(null);
+        } catch(IllegalArgumentException e) {
+            throw new InvalidObjectIdException("CollectionId: " + id);
```

you're still swallowing the exception. The stack trace will be gone.